### PR TITLE
keadm: support init kubeedge with package manager `pacman`

### PR DESCRIPTION
**Pull request by gccio**
**What type of PR is this?**
I work with manjaro os, a os like arch linux. I can't join the kubeedge cluster. I hope keam support my os type.

/kind feature

**What this PR does / why we need it**:
just want keadm can support my os.

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:
I believe nobody use this os for production. But I hope keadm can support it. :)

**Does this PR introduce a user-facing change?**:

```release-note
keadm can init kubeedge with package manager `pacman`.
```

### CODE REVIEW

**Comment by fisherxu** [comment]
Sorry for the delay and thanks for the enhancement, would you like to have a look at #2388, with that patch, we can detect the package manager and avoid to specify OS type :)

**Comment by gccio** [comment]
> Sorry for the delay and thanks for the enhancement, would you like to have a look at #2388, with that patch, we can detect the package manager and avoid to specify OS type :)

I have been updated it. Now support pacman package manager.

File: keadm/cmd/keadm/app/cmd/util/pacmaninstaller.go [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+
+import (
+	"fmt"
+
+	"github.com/blang/semver"
+
+	types "github.com/kubeedge/kubeedge/keadm/cmd/keadm/app/cmd/common"
+)
+
+//PacmanOS struct objects shall have information of the tools version to be installed
```
Comment (daixiang0) [inline keadm/cmd/keadm/app/cmd/util/pacmaninstaller.go:11]: Add blank after `//`, fix all.

File: keadm/cmd/keadm/app/cmd/util/common.go [hunk 2]
```diff
@@ -64,8 +64,13 @@ const (
 	latestReleaseVersionURL = "https://kubeedge.io/latestversion"
 	RetryTimes              = 5
 
-	APT string = "apt"
-	YUM string = "yum"
+	OSArchAmd64 string = "amd64"
```
Comment (daixiang0) [inline keadm/cmd/keadm/app/cmd/util/common.go:67]:
```
AMD64
ARM64
ARM32
```

File: keadm/cmd/keadm/app/cmd/util/common.go [hunk 3]
```diff
@@ -119,6 +127,8 @@ func GetOSInterface() types.OSTypeInstaller {
 		return &DebOS{}
 	case YUM:
 		return &RpmOS{}
+	case PACMAN:
+		return &PacmanOS{}
 	default:
 		fmt.Println("Failed to detect supported package manager command(apt, yum), exit")
 		panic("Failed to detect supported package manager command(apt, yum), exit")
```
Comment (daixiang0) [inline keadm/cmd/keadm/app/cmd/util/common.go:134]: Update here too.

File: keadm/cmd/keadm/app/cmd/util/pacmaninstaller.go (hunk 1 above)
Comment (gccio) [reply keadm/cmd/keadm/app/cmd/util/pacmaninstaller.go]: fixed.

File: keadm/cmd/keadm/app/cmd/util/common.go (hunk 2 above)
Comment (gccio) [reply keadm/cmd/keadm/app/cmd/util/common.go]: fixed.

File: keadm/cmd/keadm/app/cmd/util/common.go (hunk 3 above)
Comment (gccio) [reply keadm/cmd/keadm/app/cmd/util/common.go]: updated.

**Comment by daixiang0** [comment]
lgtm

**Comment by fisherxu** [comment]
@gccio Please squash the commits :)

**Comment by gccio** [comment]
> @gccio Please squash the commits :)

has been squashed.

**Comment by gccio** [comment]
any other problem?

**Comment by daixiang0** [comment]
Please update pr title and fill something under `Does this PR introduce a user-facing change`

**Comment by gccio** [comment]
Hi @fisherxu , is there any other problem?

**Comment by gccio** [comment]
> Please update pr title and fill something under `Does this PR introduce a user-facing change`

@daixiang0 Done

**Pull request merged**